### PR TITLE
fix versionize-docs.sh and mark-new-version.sh to rewrite links to release branch rather than to tag

### LINKED ISSUE #16522: links in release branch should point to release branch, rather than version tags
For example, in this .md file in the release-1.1 branch, https://github.com/kubernetes/kubernetes/blob/release-1.1/docs/user-guide/configuring-containers.md, 
the "Pod API object" link links to https://htmlpreview.github.io/?https://github.com/kubernetes/kubernetes/blob/v1.1.0/docs/api-reference/definitions.html#_v1_pod,
it should rather points to 
https://htmlpreview.github.io/?https://github.com/kubernetes/kubernetes/release-1.1/docs/api-reference/v1/definitions.html#_v1_pod

I will fix this in versionized-docs.sh.

[edit] it seems the problem is the mark-new-version.sh, https://github.com/kubernetes/kubernetes/blob/master/build/mark-new-version.sh#L143
We need rewrite the links to point to release branch before push the release branch.

cc @nikhiljindal @janetkuo @bgrant0607 

And this issue (https://github.com/kubernetes/kubernetes/pull/16486#discussion_r43350745) will be solved as well.

### PR DESCRIPTION
**Pull request by caesarxuchao**
Fix #16522.

@nikhiljindal @janetkuo @bgrant0607

### CODE REVIEW

**Comment by ihmccreery** [comment]
I don't think this is right.  We should discuss further in #16522.  We do, for official releases, (which is what `mark-new-version` is for,) point to actual tags, not the branch.

File: build/versionize-docs.sh [hunk 1]
```diff
@@ -27,11 +27,12 @@ KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
...
 NEW_VERSION=${1-}
 
-# NEW_VERSION is expected to be vMajor.Minor.Micro.
-MAJOR_AND_MINOR_VERSION=${NEW_VERSION%.*}
+# MAJOR_AND_MINOR_VERSION is expected to be something like "v1.1"
+MAJOR_AND_MINOR_VERSION="v${NEW_VERSION#release-}"
 
 if [ "$#" -lt 1 ]; then
-    echo "Usage: versionize-docs <release-version>"
+    echo "Usage: versionize-docs.sh <release-version>, e.g., versionize-docs.sh release-1.1."
```
Comment (bgrant0607) [inline build/versionize-docs.sh]: `<release-branch>` rather than `<release-version>`

File: build/versionize-docs.sh [hunk 2]
```diff
@@ -27,11 +27,12 @@ KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
 
 NEW_VERSION=${1-}
```
Comment (bgrant0607) [inline build/versionize-docs.sh]: It would be less confusing to rename this variable to RELEASE_BRANCH.

**Comment by caesarxuchao** [comment]
@bgrant0607 fixed. PTAL. Thanks.

File: build/versionize-docs.sh [hunk 3]
```diff
@@ -59,11 +60,11 @@ done
 for doc in "${md_files[@]}"; do
   $SED -ri \
       -e '/<!-- BEGIN STRIP_FOR_RELEASE -->/,/<!-- END STRIP_FOR_RELEASE -->/d' \
-      -e "s|(releases.k8s.io)/[^/]+|\1/${NEW_VERSION}|g" \
+      -e "s|(releases.k8s.io)/[^/]+|\1/${RELEASE_BRANCH}|g" \
       "${doc}"
 
   # Replace /HEAD in html preview links with /NEW_VERSION.
```
Comment (bgrant0607) [inline build/versionize-docs.sh]: s/NEW_VERSION/RELEASE_BRANCH/

**Comment by bgrant0607** [comment]
LGTM

**Comment by bgrant0607** [comment]
Merging

**Pull request merged**